internal/indexcoord/node_manager.go

```diff
@@ -102,34 +105,35 @@ func (nm *NodeManager) AddNode(nodeID UniqueID, address string) error {
 }
 
 // PeekClient peeks the client with the least load.
-func (nm *NodeManager) PeekClient(meta Meta) (UniqueID, types.IndexNode) {
-	log.Debug("IndexCoord NodeManager PeekClient")
-
+func (nm *NodeManager) PeekClient(meta *Meta) (UniqueID, types.IndexNode) {
```

Comment:
Add RLock， since you are accessing the nm.nodeClients

Reply:
A round robin or random algorithm is recommended. It is a good practice that the first NodeID at the beginning of the loop is different each time